apps/consumer-server/src/routing/routes/third-party-auth/anon-login.ts

```diff
@@ -0,0 +1,100 @@
+import { ZKEdDSAEventTicketPCDPackage } from "@pcd/zk-eddsa-event-ticket-pcd";
+import express, { Request, Response } from "express";
+import { ApplicationContext } from "../../../types";
+
+const nullifiers = new Set<string>();
+
+/**
+ * The login checks the validity of the PCD, ensures that the ticket
+ * is indeed supported by Zupass, and that it has been signed with the correct
+ * EdDSA key. The watermark used to create the PCD and as a nonce in the
+ * authentication mechanism must be the same as the one in the current session.
+ * Once all checks are passed, a user session is created in which the watermark
+ * and nullifier are saved.
+ */
+export function anonLogin(
+  app: express.Application,
+  _context: ApplicationContext
+): void {
+  app.post("/auth/anon-login", async (req: Request, res: Response) => {
+    try {
+      if (!req.body.pcd) {
+        console.error(`[ERROR] No PCD specified`);
+
+        res.status(400).send();
+        return;
+      }
+
+      const pcd = await ZKEdDSAEventTicketPCDPackage.deserialize(req.body.pcd);
+
+      if (!(await ZKEdDSAEventTicketPCDPackage.verify(pcd))) {
+        console.error(`[ERROR] ZK ticket PCD is not valid`);
+
+        res.status(401).send();
+        return;
+      }
+
+      if (pcd.claim.watermark.toString() !== req.session.nonce) {
+        console.error(`[ERROR] PCD watermark doesn't match`);
+
+        res.status(401).send();
+        return;
+      }
+
+      if (!pcd.claim.nullifierHash) {
+        console.error(`[ERROR] PCD ticket nullifier has not been defined`);
+
+        res.status(401).send();
+        return;
+      }
+
+      if (nullifiers.has(pcd.claim.nullifierHash)) {
+        console.error(`[ERROR] PCD ticket has already been used`);
+
+        res.status(401).send();
+        return;
+      }
+
+      // It fetches the ticket types from Zupass to verify whether the PCD ticket
+      // is indeed among the supported tickets and has been signed with the key from the Zupass server.
+      const response = await fetch(
+        // eslint-disable-next-line turbo/no-undeclared-env-vars
+        `${process.env.ZUPASS_API}/issue/known-ticket-types`
+      );
+      const { value } = await response.json();
```

Comment:
cc @ichub I think for these kinds of calls we want to use the APIs defined in `passport-interface` rather than plain `fetch()` calls. This also gives you TypeScript types for the results.

Reply:
I didn't know the APIs were defined there! It's using `passport-interface` now.